libs/remix-simulator/src/VmProxy.ts

```diff
@@ -133,23 +135,21 @@ export class VmProxy {
     this.storageCache[this.processingHash] = {}
     this.storageCache['after_' + this.processingHash] = {}
     if (data.to) {
-      try {
-        const storage = await this.vm.stateManager.dumpStorage(data.to)
-        this.storageCache[this.processingHash][tx['to']] = storage
-        this.lastProcessedStorageTxHash[tx['to']] = this.processingHash
-      } catch (e) {
-        console.log(e)
-      }
+      ((processingHash, processingAccount, processingAddress, self) => {
```

Comment:
any specific reason to use anonymous method?

Reply:
yes, the root function is a callback, called multiple times. `self.stateCopy.dumpStorage(account)` take some time to resolve, I want to make sure that there's no interference between multiple call to `dumpStorage`, using enclosure allows a clear separation.